content/blog/2020-09-17-delivering-learnr-tutorials-in-a-package/index.Rmarkdown

```diff
@@ -0,0 +1,322 @@
+
+Putting your learnr tutorial in a package is also free-- and always will be, which means you can also sidestep costs associated with hosting services! Your package doesn't have to be on CRAN, and it doesn't have to contain functions or data (but it can if you want!).
+
+Never made a package before? No sweat! In this post, we'll guide you through the process.
+
+## How to make a package for a learnr tutorial
+
+```{r, out.width = "100%", echo = FALSE}
+knitr::include_graphics("pkg-illo.jpg")
+```
+
+
+We will walk through the steps for creating the bare-minimum package that you can use to house and deliver your interactive learnr tutorials to your users.
+
+Our final goal is to create a package that is "downloadable" by users, which means this package needs a GitHub repository so that it can be easily shared. If students wanted to, they’d be able to find the source code and solutions in your GitHub repository (or, with some spelunking, in the folder that houses all their R packages locally). But this would take some extra effort on the part of students.
+
+If a solution like this still works for your purposes, then read onwards!
+
+
+### Pre-requisites
+
+We assume the following:
+
+* You have a GitHub account (but you don't need to be more than a GitHub novice)
+* You are comfortable with the basics of using R and RStudio
+* You understand how R projects work
+* You are already familiar with creating a regular learnr tutorial
+
+
+
+### Getting set up
+
+Let's make sure we have all the pieces in place to create a learnr tutorial package.
+
+
+#### Packages
+
+Make sure you've installed and loaded the following packages:
+
+```{r, eval = FALSE}
+#install.packages(c("usethis", "devtools", "roxygen2"))
+library(usethis)
+library(roxygen2)
+library(devtools)
+library(learnr)
+```
+
+
+#### Create a GitHub repository
+
+If you're already familiar with the ins and outs of connecting an R Project to a GitHub repository, you can skip this section.
+We recommend following the tips on [Happy Git with R](https://happygitwithr.com/new-github-first.html#make-a-repo-on-github-2) and starting in GitHub before switching to RStudio.
+
+1. [**Create a new repository on GitHub**](https://happygitwithr.com/new-github-first.html#make-a-repo-on-github-2) for your work.
+
+    * The repository name you choose will also end up being the name of your package, so choose wisely! For the examples in this guide, we'll go with "tutorialpackage".
+
+1. **Copy the repository URL** to your clipboard.
+
+    * Do this by clicking the green Clone or Download button.
+    * Copy the HTTPS URL (looks like: `https://github.com/{yourname}/{yourrepo}.git`).
+
+    * **Or**, copy the SSH URL if you chose to set up SSH keys (looks like: `git@github.com:{yourname}/{yourrepo}.git`).
```

Comment:
I think we can safely say that if they don't know what SSH is, use the HTTPS

Reply:
Do you mean strike the SSH URL step altogether?

Reply:
```suggestion
    * **Or**, copy the SSH URL if you chose to set up SSH keys (looks like: `git@github.com:{yourname}/{yourrepo}.git`). (Note: we strongly recommend that you choose HTTPS over SSH if you don't know about SSH keys).
```

Reply:
Yeah, I think the current recommendation is to use HTTPS, I'd not even mention SSH here (if people know it, they know it anyway)